### PR TITLE
Merge code from Launchpad

### PR DESCRIPTION
**Pull request by clonedagain**
It seems the launchpad code was not yet merged here, so I propose we do it now.

### CODE REVIEW

**Comment by yvaucher** [comment]
:-1: To fix added modules have to go in `__unported__`

**Comment by alexis-via** [comment]
Cool, thanks for the initiative @clonedagain 
@yvaucher what do you mean by "to fix added modules" ? If there are some modules to fix, we should fix them, not put them in unported !

**Comment by alexis-via** [comment]
The 6 modules from LP are OK and the revision history is kept :+1: 
@clonedagain Could you take 2 minutes to fix the flake8 warnings and then we'll merge it.

**Comment by clonedagain** [comment]
@yvaucher: since this is a direct import of https://code.launchpad.net/~openerp-fr-core-editors/openerp-french-localization/8.0, I expect the modules have already migrated. If they have not, please confirm it and I'll move them to `__unported__`.

**Comment by clonedagain** [comment]
@alexis-via I committed pep8 changes, please let me know if it still fails on Travis.

**Comment by alexis-via** [comment]
@clonedagain I confirm that the modules of LP have all been ported to v8. Thanks for the PEP8 fixes.

**Comment by yvaucher** [comment]
Ok I thought the following error was coming from unported module:

```
except_orm: ('Error', u"You try to install module 'account_bank_statement_import_fr_cfonb' that depends on module 'account_bank_statement_import'.\nBut the latter module is not available in your system.")
```

`account_bank_statement_import` should be resolvable by adding dependency one its OCA repository in `.travis.yml`:

```
 - git clone https://github.com/OCA/bank-statement-import ${HOME}/bank-statement-import -b ${VERSION}
```

**Comment by clonedagain** [comment]
@yvaucher I've added the line you suggested, is it OK ? Also, I noticed this line does not contain the version, is it a problem? :

```
   - git clone https://github.com/OCA/maintainer-quality-tools.git ${HOME}/maintainer-quality-tools
```

**Comment by yvaucher** [comment]
@clonedagain It can be fixed to r2 if you wish to have a stable CI tool which won't be alterated by an update. This is not mandatory. (maintainer-quality-tools is non odoo version specific.)

For travis we progress :smiley: 

Now let's get that `account_fiscal_position_rule` from https://github.com/OCA/account-fiscal-rule

File: account_bank_statement_import_fr_cfonb/__openerp__.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    'description': """
+Module to import French bank statements in CFONB format
+=======================================================
+
+This module allows you to import the text-based French
+CFONB files in Odoo as bank statements. It depends on
+the module *account_bank_statement_import* which is
+currently available in the master branch of Odoo (future
+v9), but you can backport it to v8 with this patch:
+http://people.via.ecp.fr/~alexis/account_bank_statement_import-80-backport.diff
```
Comment (legalsylvain) [inline account_bank_statement_import_fr_cfonb/__openerp__.py:39]: Can you remove this obsolete comment. (The module is now available on OCA/bank-statement-import)

**Comment by legalsylvain** [comment]
What a big PR !
- account_bank_statement_import_fr_cfonb :+1:  code review. no test;
- l10n_fr_account_fiscal_position_rule :-1: 
  - `l10n_fr_account_fiscal_position_rule/settings/account.fiscal.position.rule.template.csv could not be processed:
    Line 30 : No matching record found for external id 'l10n_fr.fiscal_position_template_intraeub2b' in field 'Fiscal Position'`
  - import problem to fix in Travis;
- l10n_fr_state, l10n_fr_department : :+1:
  Note : We have to plan the changes of 'state' in France. (Some States will be merged soon).
- l10n_fr_base_location_geonames_import :-1: 
  - Unmet dependencies to https://github.com/OCA/partner-contact.git#8.0.
  - Unmet external dependencies to fix in Travis : unicodecsv;
- l10n_fr_fec : :-1: 
  - Unmet external dependencies to fix in Travis : unicodecsv;

Regards.

**Comment by alexis-via** [comment]
I just tested the installation of l10n_fr_account_fiscal_position_rule on v8 and it works well. There is no problem with 'l10n_fr.fiscal_position_template_intraeub2b' : this XMLID exists and it declared in addons/l10n_fr/fr_fiscal_templates.xml line 16.
So the issue is a Travis issue, not an issue in the code from LP.

**Comment by legalsylvain** [comment]
Hi,
Sorry if I was not clear. 'l10n_fr_account_fiscal_position_rule' doesn't depend on 'l10n_fr'. That is the issue.
For me, still needs fixing.
Regards.

**Comment by clonedagain** [comment]
Actually I was hoping to import the existing branch more or less as-is, considering that it used to be the reference branch. Now if you wish to review each module, would you prefer me to one PR per module instead?

**Comment by legalsylvain** [comment]
No. I don't think that is necessary, because it is just ported work from launchpad.
But you have to fix : 
- dependencies in travis file; (extra repository & python librairies);
-  dependencies on 'l10n_fr' in openerp.py file;

(and better if you can remove obsolete comment as said in a previous message)

For the next PR, yes it's better to do a PR per module.

Regards.

**Comment by clonedagain** [comment]
@legalsylvain: @alexis-via provided me with patches for what you requested but it still bombs on Travis. 
`openerp_template openerp.modules.module: No module named unidecode`
Any idea?

File: .travis.yml [hunk 2]
```diff
@@ -13,8 +13,12 @@ virtualenv:
 
 install:
   - git clone https://github.com/OCA/maintainer-quality-tools.git ${HOME}/maintainer-quality-tools
+  - git clone https://github.com/OCA/bank-statement-import.git ${HOME}/bank-statement-import -b ${VERSION}
+  - git clone https://github.com/OCA/partner-contact.git ${HOME}/partner-contact
+  - git clone https://github.com/OCA/account-fiscal-rule.git ${HOME}/account-fiscal-rule
   - export PATH=${HOME}/maintainer-quality-tools/travis:${PATH}
   - travis_install_nightly
+  - sudo pip install unicodecsv
```
Comment (yvaucher) [inline .travis.yml]: I guess `unidecode` install is missing here

It is imported here:
https://github.com/numerigraphe/l10n-france/blob/8.0-launchpad/l10n_fr_base_location_geonames_import/wizard/geonames_import.py#L24

Thought dependency is marked as `unicodecsv`
https://github.com/numerigraphe/l10n-france/blob/8.0-launchpad/l10n_fr_base_location_geonames_import/__openerp__.py#L50

**Comment by clonedagain** [comment]
OK, thanks, now it's green.

**Comment by yvaucher** [comment]
Well done :+1:

**Pull request merged**